include/common/core/tool_detection.hpp

```diff
@@ -0,0 +1,189 @@
+#pragma once
+#include <cstdint>
+#include <ranges>
+#include <span>
+
+#include "can/core/ids.hpp"
+
+namespace tool_detection {
+
+// A quick specialization that is syntactically meaningless but a check
+// to the programmer
+using millivolts_t = uint16_t;
+
+// Contains the bounds for value checking
+struct ToolCheckBounds {
+    millivolts_t upper;
+    millivolts_t lower;
+};
+
+// Which carrier a device is attached to
+enum Carrier : uint8_t {
+    GRIPPER_CARRIER = 0x00,
+    Z_CARRIER = 0x01,
+    A_CARRIER = 0x02,
+    UNKNOWN = 0x03,
+};
+
+// A full specification of an attached tool, including its bounds
+struct Tool {
+    can_ids::ToolType tool_type;
+    Carrier tool_carrier;
+    ToolCheckBounds bounds;
+    [[nodiscard]] auto within_bounds(millivolts_t reading) const -> bool {
+        return ((reading >= this->bounds.lower) &&
+                (reading < this->bounds.upper));
+        return false;
```

Comment:
Why two returns? Am I crazy?

Reply:
What how did that get there why did no tooling care about it gah